### PR TITLE
Fix emphasis with underscore inside word

### PR DESCRIPTION
**Pull request by mccraveiro**
**Marked version:** master branch

**Markdown flavor:** CommonMark

## Description

- Related to #1216 

## Input

```markdown
пристаням_стремятся_
```

## Expectation

```html
<p>пристаням_стремятся_</p>
```

## Result

```html
<p>пристаням<em>стремятся</em></p>
```

## What was attempted

Fix emphasis with underscore inside word by implementing ["left flanking delimiter"](https://spec.commonmark.org/0.28/#left-flanking-delimiter-run)

## Contributor

- [x] Test(s) exist to ensure functionality and minimize regression (if no tests added, list tests covering this PR); or,
- [ ] no tests required for this PR.
- [ ] If submitting new feature, it has been documented in the appropriate places.

## Committer

In most cases, this should be a different person than the contributor.

- [ ] Draft GitHub release notes have been updated.
- [ ] CI is green (no forced merge required).
- [ ] Merge PR

### CODE REVIEW

**Comment by mccraveiro** [comment]
Regex lookbehind assertions are not available on node v4 😞 
By the way, shouldn't we move to node v6 as node v4 end-of-life was on April?

https://github.com/nodejs/Release#end-of-life-releases

File: lib/marked.js [hunk 1]
```diff
@@ -541,12 +541,12 @@ var inline = {
   link: /^!?\[(label)\]\(href(?:\s+(title))?\s*\)/,
   reflink: /^!?\[(label)\]\[(?!\s*\])((?:\\[\[\]]?|[^\[\]\\])+)\]/,
   nolink: /^!?\[(?!\s*\])((?:\[[^\[\]]*\]|\\[\[\]]|[^\[\]])*)\](?:\[\])?/,
-  strong: /^__([^\s])__(?!_)|^\*\*([^\s])\*\*(?!\*)|^__([^\s][\s\S]*?[^\s])__(?!_)|^\*\*([^\s][\s\S]*?[^\s])\*\*(?!\*)/,
+  strong: /^__([^\s_])__(?!_)|^\*\*([^\s])\*\*(?!\*)|^__([^\s][\s\S]*?[^\s])__(?!_)|^\*\*([^\s][\s\S]*?[^\s])\*\*(?!\*)/,
```
Comment (UziTech) [inline lib/marked.js]:
```suggestion
  strong: /^__([^\s_])__(?!_)|^\*\*([^\s*])\*\*(?!\*)|^__([^\s][\s\S]*?[^\s])__(?!_)|^\*\*([^\s][\s\S]*?[^\s])\*\*(?!\*)/,
```

will pass 416 as well

Comment (mccraveiro) [reply lib/marked.js]: Shouldn't we fix this on another PR? I was focusing this PR on words with underscore inside them. However, I had to also fix this "strong" tag so the existing tests still works.

Comment (UziTech) [reply lib/marked.js]: I would rather have less PRs to review.

Comment (mccraveiro) [reply lib/marked.js]: @UziTech I'll take that in consideration from now on :)

**Comment by UziTech** [comment]
I'm ok with removing node v4 support.

**Comment by mccraveiro** [comment]
It seems that Regex Lookbehind is behind a flag on node v6. 
I'll try to solve this problem with another regex pattern.

**Comment by mccraveiro** [comment]
@UziTech By the time being I've moved the last 2 commits to another PR so we can move on https://github.com/markedjs/marked/pull/1400

**Comment by mccraveiro** [comment]
It's way harder to implement without lookbehind assertions. I'll close this PR for now.

**Pull request closed without merging**